### PR TITLE
Pass along realtime-js options

### PR DESCRIPTION
**Pull request by quicksnap**
## What kind of change does this PR introduce?

Feature

## What is the new behavior?

Passes along constructor options to realtime-js

Depends on https://github.com/supabase/realtime-js/pull/95 before merge.

### CODE REVIEW

File: src/lib/types.ts [hunk 1]
```diff
@@ -23,6 +25,11 @@ export type SupabaseClientOptions = {
    * A storage provider. Used to store the logged in session.
    */
   localStorage?: Storage
+
+  /**
+   * Options passed to the realtime-js instance
+   */
+  realtimeOptions?: RealtimeClientOptions
```
Comment (w3b6x9) [inline src/lib/types.ts]: @quicksnap regarding earlier comment recommending change to `realtime`:

```suggestion
  realtime?: RealtimeClientOptions
```

Comment (quicksnap) [reply src/lib/types.ts]: updated!

File: src/SupabaseClient.ts [hunk 2]
```diff
@@ -59,7 +60,7 @@ export default class SupabaseClient {
     this.schema = settings.schema
 
     this.auth = this._initSupabaseAuthClient(settings)
-    this.realtime = this._initRealtimeClient()
+    this.realtime = this._initRealtimeClient(settings.realtimeOptions)
```
Comment (w3b6x9) [inline src/SupabaseClient.ts]: @quicksnap let's change `realtimeOptions` to `realtime` to keep it simple:

```suggestion
    this.realtime = this._initRealtimeClient(settings.realtime)
```

File: src/SupabaseClient.ts [hunk 3]
```diff
@@ -154,9 +155,10 @@ export default class SupabaseClient {
     })
   }
 
-  private _initRealtimeClient() {
+  private _initRealtimeClient(options?: RealtimeClientOptions) {
     return new RealtimeClient(this.realtimeUrl, {
-      params: { apikey: this.supabaseKey },
+      ...options,
+      params: { apikey: this.supabaseKey, ...options?.params },
```
Comment (w3b6x9) [inline src/SupabaseClient.ts]: @quicksnap let's flip the order so `apikey` is set later to bar users from overwriting via realtime params

File: src/SupabaseClient.ts (hunk 2 above)
Comment (w3b6x9) [reply src/SupabaseClient.ts]: @quicksnap can you please also add this to function doc: `* @param options.realtime...`

Comment (quicksnap) [reply src/SupabaseClient.ts]: All done!

File: src/SupabaseClient.ts (hunk 3 above)
Comment (quicksnap) [reply src/SupabaseClient.ts]: Done!

**Comment by quicksnap** [comment]
Should be good to go

**Comment by w3b6x9** [comment]
@quicksnap yep, let me do a release of `realtime-js`, version bump it here, and then we can merge this

**Comment by quicksnap** [comment]
Should be good to go now

**Review by w3b6x9 — APPROVED** [review]
@quicksnap LGTM!

**Pull request merged**